solr/core/src/java/org/apache/solr/cloud/SyncStrategy.java

```diff
@@ -361,12 +360,11 @@ private void requestRecovery(
               RequestRecovery recoverRequestCmd = new RequestRecovery();
               recoverRequestCmd.setAction(CoreAdminAction.REQUESTRECOVERY);
               recoverRequestCmd.setCoreName(coreName);
-
               try (SolrClient client =
-                  new HttpSolrClient.Builder(baseUrl)
-                      .withHttpClient(SyncStrategy.this.client)
+                  new Http2SolrClient.Builder(baseUrl)
```

Comment:
> Should this be merged or should it be pending some authentication/authorization fix?

Yes wherever we are re-creating the Http2SolrClient and in SyncStrategy we are -- because of custom timeouts. This code for listener factory seems a broken window to me as there is no way to enforce it and can be easily missed in future. @dsmiley We had discussion regarding the timeouts handling at request level then we don't need to re-create the client.

Other option that I am also in favour of is to copy variable via calling `withHttpClient(solrClient)` as we are already doing with other configuration.

https://github.com/apache/solr/blob/6af52f33ac78155523ec6fc610c563127b81c139/solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java#L1033-L1058

Reply:
Shouldn't withHttpClient copy *everything* over, including the ListenerFactory?  It's worthy of its own JIRA/PR.

Reply:
@dsmiley #2467 is for that one.